model/metadata/metadata.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import "github.com/prometheus/prometheus/model/textparse"
+
+// Metadata stores a series' metadata information.
+type Metadata struct {
+	Type textparse.MetricType
+	Unit string
+	Help string
+}
+
+var emptyMetadata = Metadata{}
+
+func EmptyMetadata() Metadata {
+	return emptyMetadata
+}
```

Comment:
tiny nit: I don't know if functions to return a zero value are really common, I'd just have the caller use `metadata.Metadata{}` if they needed it (which is also shorter than calling `metadata.EmptyMetadata()`.

Don't have a function for this, have the caller create the zero struct manually if they need it.

Reply:
I don't really like it too tbh, but I saw it being used in `EmptySeriesSet`.  Fortunately with this refactor it's only used in one place, so I'll get rid of it.